glib/src/property.rs

```diff
@@ -37,6 +38,9 @@ impl<T: Property> Property for Mutex<T> {
 impl<T: Property> Property for RwLock<T> {
     type Value = T::Value;
 }
+impl<T: Property> Property for ThreadGuard<T> {
+    type Value = T::Value;
```

Comment:
Value should probably be Self here, probably a Boxed object?

Reply:
No, because `Property::Value` is used to retrieve the innermost type, the one with `HasParamSpec` trait.

We can't put `type Value = Self` because `Self` doesn't implement `HasParamSpec`